File: totaleconomy-paper/src/main/java/com/ericgrandt/totaleconomy/impl/PaperPlayer.java
package com.ericgrandt.totaleconomy.impl;

import com.ericgrandt.totaleconomy.model.TEPlayer;
import net.kyori.adventure.text.Component;
import org.bukkit.entity.Player;

import java.util.UUID;

public class PaperPlayer implements TEPlayer {
    private final Player player;

    public PaperPlayer(Player player) {
        this.player = player;
    }

    @Override
    public UUID uniqueId() {
        return player.getUniqueId();
    }

    @Override
    public String name() {
        return player.getName();
    }

    @Override
    public void sendMessage(Component message) {
        player.sendMessage(message);
    }
}

File: totaleconomy-paper/src/main/kotlin/com/ericgrandt/totaleconomy/impl/PaperPlayer.kt
package com.ericgrandt.totaleconomy.impl

import com.ericgrandt.totaleconomy.model.Player
import net.kyori.adventure.text.Component
import java.util.UUID
import org.bukkit.entity.Player as PlayerEntity

class PaperPlayer(
    private val player: PlayerEntity,
) : Player {
    override val uniqueId: UUID = player.uniqueId

    override val name: String = player.name

    override fun sendMessage(message: Component) {
        player.sendMessage(message)
    }
}
